pandas/_libs/src/datetime.pxd

```diff
@@ -161,3 +162,23 @@ cdef inline int64_t _date_to_datetime64(object val,
     dts.hour = dts.min = dts.sec = dts.us = 0
     dts.ps = dts.as = 0
     return pandas_datetimestruct_to_datetime(PANDAS_FR_ns, dts)
+
+
+cdef inline check_dts_bounds(pandas_datetimestruct *dts):
+    cdef:
```

Comment:
better to make this a bint, and just return True/False.

Then make another function which *calls* this one which actually raises the error.

Reply:
The longer-term solution that I've implemented but not PRed is to move a bunch of datetime.pxd into a pyx file where OutOfBoundsDatetime can be defined.  This also ends up clearing out _a lot_ of the setup.py dependencies (orthogonal to this discussion, but still).

Is returning a bint actually any simpler than re-raising?  It isn't obvious whether True means "error state is True" or "everything is OK is True", whereas raising is unambiguous.

That said, I'll do it your way.

Reply:
yeah its ok, just don't like to muddle even more.